morphic.js

```diff
@@ -9403,16 +9439,34 @@ MenuItemMorph.prototype.mouseEnter = function () {
         this.image = this.highlightImage;
         this.changed();
     }
+    if (this.doubleClickAction instanceof MenuMorph) {
+        var w = this.world();
+        this.doubleClickAction.popup(w,
+                                     new Point(this.parent.bounds.right()-10,
+                                     w.hand.position().y-10),
+                                     true);
+        if (!contains(this.parent.children, this.doubleClickAction))
+            this.parent.addChild(this.doubleClickAction);
+    }
     if (this.hint) {
         this.bubbleHelp(this.hint);
     }
 };
 
 MenuItemMorph.prototype.mouseLeave = function () {
-    if (!this.isListItem()) {
+    if (this.doubleClickAction instanceof MenuMorph) {
```

Comment:
I find this very confusing to read. I think it would be clearer if you'd just add a property named ```submenu``` or so, instead of overloading a property that's meant for double clicks